src/grid/becke.py

```diff
@@ -78,37 +117,29 @@ def _switch_func(x, order=3):
             x = 1.5 * x - 0.5 * x ** 3
         return x
 
-    @staticmethod
-    def generate_becke_weights(
-        points, radii, atom_coors, *, select=None, pt_ind=None, order=3
-    ):
-        """Calculate Becke weights of points for select atom.
+    def generate_weights(self, points, *, select=None, pt_ind=None):
```

Comment:
Could become `__call__`

Reply:
Sounds good to me. Then the `HirshfeldWeights` class would have a `__call__` method as well... I was also thinking to put the chunking hack for computing Becke weight in the `__call__` function too so that it would become simpler...

Reply:
agreed